### PR TITLE
noderesourcetopologies: address pending minor cleanups

### PR DESCRIPTION
**Pull request by ffromani**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
During review of PRs related to NodeResourceTopology (#474 #439, #437), we collected a list of cleanups and nice-to-have which I promised to fix. Here's the PR addressing them.

#### Which issue(s) this PR fixes:
Fixes N/A

#### Special notes for your reviewer:
minor/suggestion: if we merge this PR, commits can (and probably should?) be squashed together.

#### Does this PR introduce a user-facing change?
```release-note
none
```

### CODE REVIEW

**Review by swatisehgal — APPROVED** [review]
Thanks for submitting this cleanup PR!
/lgtm

/assign @Huang-Wei 
for approval

**Comment by swatisehgal** [comment]
/hold

as @PiotrProkop has identified that the handler signature needs improvement

Please feel free to unhold when that has been addressed.

File: pkg/noderesourcetopology/score.go [hunk 1]
```diff
@@ -74,16 +74,21 @@ func (tm *TopologyMatch) Score(ctx context.Context, state *framework.CycleState,
...
+	policyName := nodeTopology.TopologyPolicies[0]
+	handler, ok := tm.scoringHandlers[topologyv1alpha1.TopologyManagerPolicy(policyName)]
+	if !ok {
+		klog.V(4).InfoS("policy handler not found", "policy", policyName)
+		return 0, nil
+	}
+
+	// calculates the fraction of requested to capacity per each numa-node.
+	// return the numa-node with the minimal score as the node's total score
+	return handler(pod, nodeTopology.Zones, tm.scorerFn, tm.resourceToWeightMap)
```
[the remark this answers is not in the document]
Comment (ffromani) [reply pkg/noderesourcetopology/score.go:91]: right, rebase glitch. Thanks for spotting!

File: pkg/noderesourcetopology/score.go [hunk 2]
```diff
@@ -74,16 +74,21 @@ func (tm *TopologyMatch) Score(ctx context.Context, state *framework.CycleState,
...
+
+	policyName := nodeTopology.TopologyPolicies[0]
+	handler, ok := tm.scoringHandlers[topologyv1alpha1.TopologyManagerPolicy(policyName)]
+	if !ok {
+		klog.V(4).InfoS("policy handler not found", "policy", policyName)
+		return 0, nil
+	}
+
+	// calculates the fraction of requested to capacity per each numa-node.
+	// return the numa-node with the minimal score as the node's total score
```
Comment (PiotrProkop) [inline pkg/noderesourcetopology/score.go:90]: I think the comment is a bit off, cause now we also have multi zones scoringPolicies

Comment (ffromani) [reply pkg/noderesourcetopology/score.go:90]: right again. Let me just remove this comment.

**Review by swatisehgal — COMMENTED** [review]
/lgtm

**Comment by ffromani** [comment]
Thanks @PiotrProkop and @swatisehgal for the lightining fast review!

**Comment by ffromani** [comment]
note: if we merge this PR, commits can (and probably should?) be squashed together.

File: pkg/noderesourcetopology/filter.go [hunk 3]
```diff
@@ -207,17 +207,22 @@ func (tm *TopologyMatch) Filter(ctx context.Context, cycleState *framework.Cycle
...
-		if handler, ok := tm.filterHandlers[topologyv1alpha1.TopologyManagerPolicy(policyName)]; ok {
-			if status := handler(pod, nodeTopology.Zones, nodeInfo); status != nil {
-				tm.nrtCache.NodeMaybeOverReserved(nodeName, pod)
-				return status
-			}
-		} else {
-			klog.V(5).InfoS("Policy handler not found", "policy", policyName)
-		}
+	if len(nodeTopology.TopologyPolicies) == 0 {
+		klog.V(2).InfoS("Cannot determine policy", "node", nodeName)
```
Comment (Huang-Wei) [inline pkg/noderesourcetopology/filter.go:211]: Usually, we don't use V(2) :) V(3) or V(4) is more conventional given it's not a critical issue I suppose.

**Comment by Huang-Wei** [comment]
> note: if we merge this PR, commits can (and probably should?) be squashed together.

In the before, we had to request the PR author to squash commits, to have a clean commit history. Now that's optional, we can tag a particular label to save that additional round of "ping-pong".

/label tide/merge-method-squash
/approve

Thanks @ffromani !

**Pull request merged**